xdsl/dialects/riscv.py

```diff
@@ -183,6 +183,39 @@ class NullaryOperation(IRDLOperation, ABC):
     pass
 
 
+class CsrSwapOperation(IRDLOperation, ABC):
+    """
+    A base class for RISC-V operations performing a swap to/from a CSR.
+
+    The 'writeonly' attribute controls the actual behaviour of the operation:
+    * when True, the operation writes the rs value to the CSR but never reads it and
```

Comment:
We can add a `verify_` method that checks this, there are a few examples in the codebase

Reply:
Do we allow for boolean attributes (ones that do not take a value but that can be just present or not). Maybe worth checking with @math-fehr?

Reply:
Yes, for boolean attributes, we should use `UnitAttr` instead.
Here is an example on how to define it here: https://github.com/xdslproject/xdsl/blob/4f6b830d924fa165753a52de3c874690b87b11c9/xdsl/dialects/gpu.py#L132
In the attribute dictionary, you don't include the attribute if it is unset, and you can write `{writeonly, /* other attributes */}` if it is set (you don't have to write `= true` or anything like that).

Reply:
Thanks for the suggestions, I think it turned out pretty good looking with unit attributes for flags.